packages/firestore/src/local/indexeddb_persistence.ts

```diff
@@ -191,43 +188,32 @@ export class IndexedDbPersistence implements Persistence {
    */
   static MAIN_DATABASE = 'main';
 
-  static async createIndexedDbPersistence(
-    persistenceKey: string,
-    clientId: ClientId,
-    platform: Platform,
-    queue: AsyncQueue,
-    serializer: JsonProtoSerializer,
-    lruParams: LruParams
-  ): Promise<IndexedDbPersistence> {
-    const persistence = new IndexedDbPersistence(
-      persistenceKey,
-      clientId,
-      platform,
-      queue,
-      serializer,
-      lruParams
-    );
-    await persistence.start();
-    return persistence;
-  }
+  static async createIndexedDbPersistence(options: {
+    allowTabSynchronization: boolean;
+    persistenceKey: string;
+    clientId: ClientId;
+    platform: Platform;
+    lruParams: LruParams;
+    queue: AsyncQueue;
+    serializer: JsonProtoSerializer;
+    sequenceNumberSyncer: SequenceNumberSyncer;
+  }): Promise<IndexedDbPersistence> {
+    if (!IndexedDbPersistence.isAvailable()) {
+      throw new FirestoreError(
+        Code.UNIMPLEMENTED,
+        UNSUPPORTED_PLATFORM_ERROR_MSG
+      );
+    }
 
-  static async createMultiClientIndexedDbPersistence(
-    persistenceKey: string,
-    clientId: ClientId,
-    platform: Platform,
-    queue: AsyncQueue,
-    serializer: JsonProtoSerializer,
-    lruParams: LruParams,
-    multiClientParams: MultiClientParams
-  ): Promise<IndexedDbPersistence> {
     const persistence = new IndexedDbPersistence(
-      persistenceKey,
-      clientId,
-      platform,
-      queue,
-      serializer,
-      lruParams,
-      multiClientParams
+      options.allowTabSynchronization,
+      options.persistenceKey,
+      options.clientId,
+      options.platform,
+      options.lruParams,
+      options.queue,
+      options.serializer,
+      options.sequenceNumberSyncer
```

Comment:
It seems slightly odd and arbitrary for createIndexedDbPersistence() to take an options object but then we flatten that into separate arguments when calling the IndexedDbPersistence constructor.

I don't care very strongly, but it seems like it'd be cleaner to either convert both to options objects or just use plain arguments for both.

Reply:
Oh, right!  That's a pretty good reason.